### PR TITLE
Provide the stream to which the WritedResult pertains when available fixes #1176

### LINKED ISSUE #1176: Add the stream name to the WriteResult
Could we have the stream name added to the writeResult structure of the client API. This would allow integration with frameworks like Akka.net

### PR DESCRIPTION
**Pull request by SeanFarrow**
fixes #1176 
Allows us to detect the stream to which a write has happened, but is not available when a transaction is committed.

### CODE REVIEW

File: src/EventStore.ClientAPI/ClientOperations/AppendToStreamOperation.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ protected override InspectionResult InspectResponse(ClientMessage.WriteEventsCom
 
         protected override WriteResult TransformResponse(ClientMessage.WriteEventsCompleted response)
         {
-            return new WriteResult(response.LastEventNumber, new Position(response.PreparePosition ?? -1, response.CommitPosition ?? -1));
+            return new WriteResult(response.LastEventNumber, new Position(response.PreparePosition ?? -1, response.CommitPosition ?? -1), this._stream);
```
Comment (pgermishuys) [inline src/EventStore.ClientAPI/ClientOperations/AppendToStreamOperation.cs]: tiny nitpick, can you possibly remove the use of `this` from `this._stream`?

**Comment by pgermishuys** [comment]
@SeanFarrow apologies for doing this to you, but you removed the wrong `this` 😄 . There is a comment on the particular line I was referring to.
Might be a good idea to revert to previous commit, add new commit, squash and force push up.

**Comment by pgermishuys** [comment]
@SeanFarrow would you be willing to squash the commits for us into a single logical commit? This is merely to keep the history looking reasonable.
Apologies for the run around! Thanks for the PR!

**Comment by pgermishuys** [comment]
Hi @SeanFarrow just following up regarding this PR. Would you mind squashing the commits into a logical commit?

**Comment by SeanFarrow** [comment]
Hi,

I’ll do this in the next few days.
Cheers
Sean.

**Comment by SeanFarrow** [comment]
Does this need more work/any chance it could be merged is not?

**Comment by pgermishuys** [comment]
@SeanFarrow I sent you an email earlier last week on the 7th of February regarding the changes requested. Did you receive it?

**Comment by SeanFarrow** [comment]
Hi,

Yes, I will be doing this tomorrow.
Kind regards
Sean.

**Comment by pgermishuys** [comment]
Closing this PR for now upon request from @SeanFarrow. The PR will be opened by him again at a later stage.

**Pull request closed without merging**